go/libraries/doltcore/sqle/log_table.go

```diff
@@ -0,0 +1,138 @@
+// Copyright 2019 Liquidata, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package sqle
+
+import (
+	"io"
+
+	"github.com/src-d/go-mysql-server/sql"
+
+	"github.com/liquidata-inc/dolt/go/libraries/doltcore/doltdb"
+	"github.com/liquidata-inc/dolt/go/libraries/doltcore/env"
+	"github.com/liquidata-inc/dolt/go/libraries/doltcore/env/actions"
+)
+
+const (
+	// LogTableName is the system table name
+	LogTableName = "__log__"
+)
+
+var _ sql.Table = (*LogTable)(nil)
+
+// LogTable is a sql.Table implementation that implements a system table which shows the dolt commit log
+type LogTable struct {
+	dEnv *env.DoltEnv
+}
+
+// NewLogTable creates a LogTable
+func NewLogTable(dEnv *env.DoltEnv) *LogTable {
+	return &LogTable{dEnv: dEnv}
+}
+
+// Name is a sql.Table interface function which returns the name of the table which is defined by the constant
+// LogTableName
+func (dt *LogTable) Name() string {
+	return LogTableName
+}
+
+// String is a sql.Table interface function which returns the name of the table which is defined by the constant
+// LogTableName
+func (dt *LogTable) String() string {
+	return LogTableName
+}
+
+// Schema is a sql.Table interface function that gets the sql.Schema of the log system table.
+func (dt *LogTable) Schema() sql.Schema {
+	return []*sql.Column{
+		{Name: "commit_hash", Type: sql.Text, Source: LogTableName, PrimaryKey: true},
+		{Name: "committer", Type: sql.Text, Source: LogTableName, PrimaryKey: false},
+		{Name: "email", Type: sql.Text, Source: LogTableName, PrimaryKey: false},
+		{Name: "date", Type: sql.Text, Source: LogTableName, PrimaryKey: false},
+		{Name: "message", Type: sql.Text, Source: LogTableName, PrimaryKey: false},
+	}
+}
+
+// Partitions is a sql.Table interface function that returns a partition of the data.  Currently the data is unpartitioned.
+func (dt *LogTable) Partitions(*sql.Context) (sql.PartitionIter, error) {
+	return &doltTablePartitionIter{}, nil
+}
+
+// PartitionRows is a sql.Table interface function that gets a row iterator for a partition
+func (dt *LogTable) PartitionRows(sqlCtx *sql.Context, part sql.Partition) (sql.RowIter, error) {
+	return NewLogItr(sqlCtx, dt.dEnv)
+}
+
+// LogItr is a sql.RowItr implementation which iterates over each commit as if it's a row in the table.
+type LogItr struct {
+	commits []*doltdb.Commit
```

Comment:
How big can this get in practice at this point? Do we not have an iterated version, rather than loading them all into memory at once?

Reply:
Don't have anything like that atm.  Can work on that when it becomes an issue.